analyzer/src/main/kotlin/managers/NPM.kt

```diff
@@ -503,7 +503,11 @@ open class NPM(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConf
         }
 
         // Install all NPM dependencies to enable NPM to list dependencies.
-        run(workingDir, "install", "--ignore-scripts")
+        if (hasLockFile(workingDir) && this !is Yarn) {
```

Comment:
Instead of adding a dependency from the parent to its child class,
how about using a boolean for this, e.g. m_useNpmCi?

Reply:
https://github.com/heremaps/oss-review-toolkit/pull/1142/

Reply:
I'm not really in favor of such boolean toggles. They tend to pile up and the value itself (`true` or `false`) tells you nothing about the semantics. What if we'd want to run another NPM-specific command? Would we then also introduce a `m_useNpmFoo`? Probably not, and also consolidating such two booleans to a single `m_useNpm` boolean would not make much sense, as we can determine automatically whether we're running an NPM instance or not.

Also, there is not really a dependency being introduced: Even if the Yarn class would not inherit from the NPM class anymore this code would still work.